pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -145,18 +145,22 @@ func handleInferenceGraphRawDeployment(cl client.Client, clientset kubernetes.In
 	objectMeta, componentExtSpec := constructForRawDeployment(graph)
 
 	// create the reconciler
-	reconciler, err := raw.NewRawKubeReconciler(cl, clientset, scheme, objectMeta, &componentExtSpec, desiredSvc)
+	reconciler, err := raw.NewRawKubeReconciler(cl, clientset, scheme, objectMeta, metav1.ObjectMeta{}, &componentExtSpec, desiredSvc, nil)
 
 	if err != nil {
 		return nil, reconciler.URL, errors.Wrapf(err, "fails to create NewRawKubeReconciler for inference graph")
 	}
 	// set Deployment Controller
-	if err := controllerutil.SetControllerReference(graph, reconciler.Deployment.Deployment, scheme); err != nil {
-		return nil, reconciler.URL, errors.Wrapf(err, "fails to set deployment owner reference for inference graph")
+	for _, deployment := range reconciler.Deployment.DeploymentList {
```

Comment:
why do you need to do a for loop if you return deployment[0] in the `NewRawKubeReconciler` method?

Reply:
I think you misunderstood the logic. This is not about deployment[0]  in NewRawKubeReconciler. 
`reconciler.Deployment.DeploymentList` returns `[]*appsv1.Deployment`